### PR TITLE
Add support for darwin and linux

### PR DESCRIPTION
**Pull request by hasufell**
@hvr

### CODE REVIEW

File: autoconf-darwin/config.h [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+#define HAVE_CHECK_CRC32 1
+
+/* Define to 1 if crc64 integrity check is enabled. */
+#define HAVE_CHECK_CRC64 1
+
+/* Define to 1 if sha256 integrity check is enabled. */
+#define HAVE_CHECK_SHA256 1
+
+/* Define to 1 if you have the `clock_gettime' function. */
+#define HAVE_CLOCK_GETTIME 1
```
Comment (cartazio) [inline autoconf-darwin/config.h]: this isn't alwasy true ... on darwin

File: autoconf-darwin/config.h [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+/* Define to 1 if you have the <CommonCrypto/CommonDigest.h> header file. */
+#define HAVE_COMMONCRYPTO_COMMONDIGEST_H 1
+
+/* Define if the GNU dcgettext() function is already present or preinstalled.
+   */
+#define HAVE_DCGETTEXT 1
+
+/* Define to 1 if you have the declaration of `CLOCK_MONOTONIC', and to 0 if
+   you don't. */
+#define HAVE_DECL_CLOCK_MONOTONIC 1
```
Comment (cartazio) [inline autoconf-darwin/config.h]: def not true on older osx that lack have clock time

**Comment by hvr** [comment]
Since we can't have autoconf for windows enabled; are we *sure* these pre-computed autoconf values on OSX are *always* correct among all OSX releases?

Same goes for Linux where I'm even more sceptical.

**Comment by hasufell** [comment]
> are we sure these pre-computed autoconf values on OSX are always correct among all OSX releases?

As sure as on windows. Only time will tell.

**Comment by hasufell** [comment]
I'm going to abandon this PR in a week.

**Pull request closed without merging**